src/lib/components/setlist/AddMusiciansToSongModal.svelte

```diff
@@ -0,0 +1,176 @@
+<script lang="ts">
+	import { invalidateAll } from '$app/navigation'
+	import { db } from '$lib/firebase/client/firebase'
+	import type { Song } from '$lib/types/domain/song'
+	import type { User } from '$lib/types/domain/user'
+	import {
+		Modal,
+		Form,
+		TextInput,
+		ComboBox,
+		Grid,
+		Row,
+		Column,
+		Button
+	} from 'carbon-components-svelte'
+	import type { DropdownItem } from 'carbon-components-svelte/types/Dropdown/Dropdown.svelte'
+	import { AddAlt, CloseOutline } from 'carbon-icons-svelte'
+	import { addDoc, collection, deleteDoc, doc, getDocs, query, where } from 'firebase/firestore'
+
+	export let song: Song
+	export let open = false
+
+	export let users: User[]
+
+	let participantListItems: DropdownItem[] = users.map(
+		(user) => ({ id: user.id, text: user.name } as DropdownItem)
+	)
+
+	export let participants: {
+		participantId: string | undefined
+		instrument: string | undefined
+	}[]
+
+	let modalParticipants: {
+		participantId: string | undefined
+		instrument: string | undefined
+	}[] = [...participants, { participantId: undefined, instrument: undefined }]
+
+	/** Remove a participant from the song */
+	async function removeParticipantFromSong(song: Song, user: User, instrument: string) {
+		// Get the document reference to the song the user plays on
+		const songRef = doc(db, 'songs', song.id)
+		const playsInQuery = query(
+			collection(db, 'users', user.id, 'playsSongs'),
+			where('song', '==', songRef),
+			where('part', '==', instrument)
+		)
+		const playsInDoc = (await getDocs(playsInQuery)).docs[0].ref
+
+		// Delete the document
+		await deleteDoc(playsInDoc)
+	}
+
+	/** Update the musicians for the song. */
+	async function updateParticipantsOnSong() {
+		const definedParticipants = modalParticipants.filter(
+			({ participantId, instrument }) => participantId && instrument
+		)
+		const addedParticipants = definedParticipants.filter(
+			({ participantId, instrument }) =>
+				!participants.find((p) => p.participantId === participantId && p.instrument === instrument)
+		)
+		const deletedParticipants = participants.filter(
+			({ participantId, instrument }) =>
+				!definedParticipants.find(
+					(p) => p.participantId === participantId && p.instrument === instrument
+				)
+		)
+
+		// Add all missing musicians to the song
+		for (const { participantId, instrument } of addedParticipants) {
+			await addDoc(collection(db, 'users/' + participantId + '/playsSongs'), {
+				part: instrument,
+				song: doc(db, 'songs', song.id)
+			})
+		}
+
+		// Remove all musicians that are not in the song anymore
+		for (const { participantId, instrument } of deletedParticipants) {
+			const user = users.find((u) => u.id === participantId)
+			if (!user) console.error('User not found')
+			else await removeParticipantFromSong(song, user, instrument!)
+		}
+
+		invalidateAll()
+	}
+</script>
+
+<Modal
+	modalHeading="Add participant(s) to line-up"
```

Comment:
Since this is now more of an edit, should probably be reflected in this tile and the name of the component as well